homeassistant/components/sms/gateway.py

```diff
@@ -15,6 +18,120 @@ class Gateway:
     def __init__(self, worker, hass):
         """Initialize the sms gateway."""
         self._worker = worker
+        self._loop = asyncio.get_running_loop()
+        self._hass = hass
+
+    async def init_async(self):
+        """Initialize the sms gateway asyncronisly."""
+        try:
+            await self._worker.SetIncomingSMSAsync()
+        except gammu.ERR_NOTSUPPORTED:
+            _LOGGER.warning("Your phone does not support incoming SMS notifications!")
```

Comment:
Should this be a warning ? It seems heavy as it will be shown on every reboot. Better to hook into `hass.components.system_health.async_register_info` and register what is and what is not supported.

Reply:
Even on phones where this is supported I have seem the callback sometimes stops working for no reason at all, so I am thinking on falling back to replace the call of gammu.ReadDevice() (which is the thing that does polling and executes the callback anyway) for the code that checks for new incoming SMS.

Reply:
I could not find sample code that uses hass.components.system_health.async_register_info, can you give me an example?